app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -466,6 +516,14 @@ open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus: Bus,
   }
 
   override fun configureMapzenMap() {
+    if (locationClientManager.getClient() == null) {
```

Comment:
`updateLocation()`, `onClickStartNavigation()`, `generateRoutePreview()`, and `configureMapzenMap()` all use similar logic to check if the client is connected and if not post a `Runnable`. It could be cool to add a private method that takes the function to be run as a parameter.

```kotlin
  private fun connectAndPostRunnable(run: () -> Unit) {
    locationClientManager.connect()
    locationClientManager.addRunnableToRunOnConnect(Runnable { run })
  }
```

Then this logic would become:

```kotlin
    if (locationClientManager.getClient() == null) {
      connectAndPostRunnable { generateRoutePreview() }
      return
    }
```